### ISSUE TITLE
Support reading, writing Parquet key-value metadata

### ISSUE THREAD

**Issue by liquidcarbon** [pola-rs/polars#18550]
### Description

Hello,

Is there a way to do this in Polars, and also to read the kv metadata back?  This is DuckDB:

![image](https://github.com/user-attachments/assets/178569bb-f3d8-419b-910b-ff1540b51ada)

From the [function signature](https://github.com/pola-rs/polars/blob/main/crates/polars-python/src/dataframe/io.rs#L385) it seems like there is not.  Tried looking for it in case it's not documented but didn't find it.

**Comment by JSteilberg** [comment]
Duplicate of #5117, #15689

**Comment by liquidcarbon** [comment]
Thanks for the context - but not quite a duplicate.

Here I'm effectively asking for an extra argument to `pl.DataFrame.write_parquet`, and a similar reading function - a much smaller ask than #15689

**Comment by coastalwhite** [comment]
I don't think this should be that hard to implement, although it is not that high priority. What kind of metadata would be needed? Would it be sufficient to always read/write binary values?

**Comment by deanm0000** [comment]
I'm guessing this is doing the same thing as duckdb but [here's](https://stackoverflow.com/a/77642404/1818713) a pyarrow way to do it.

**Comment by liquidcarbon** [comment]
They're not binary, I think...? Arrow returns metadata in `pf.schema_arrow.metadata` as a dict with bytestring keys and values.
![image](https://github.com/user-attachments/assets/979efb04-ea0a-4b7e-a577-4ecc36030a5f)

**Comment by liquidcarbon** [comment]
> I'm guessing this is doing the same thing as duckdb but [here's](https://stackoverflow.com/a/77642404/1818713) a pyarrow way to do it.

[according to](https://www.linkedin.com/posts/jorritsandbrink_parquet-is-hard-to-write-%C2%B9-only-a-few-python-activity-7225901197323350017-qacK?utm_source=share&utm_medium=member_desktop) @ritchie46 Polars has its own writer, so relying on pyarrow is not the spirit :) 

![image](https://github.com/user-attachments/assets/1c974e7c-2f1a-4754-bffb-5f5fc2b98ed4)

**Comment by deanm0000** [comment]
> so relying on pyarrow is not the spirit :)

I hope they add that to polars but in the meantime wanted to share an alternative to the duck.

**Comment by cooknl** [comment]
The main use for me is to have a "unit" (think pounds, or volts, or any other arbitrary string) to describe the values in the column
